nikola/plugins/task/authors.py

```diff
@@ -0,0 +1,292 @@
+            context = {}
+            if has_authors:
+                context["title"] = kw["messages"][lang]["Authors"]
+                context["items"] = [(author, self.site.link("author", author, lang)) for author
+                                    in authors]
+                context["description"] = context["title"]
+            else:
+                context["items"] = None
+            context["permalink"] = self.site.link("author_index", None, lang)
+            context["pagekind"] = ["list", "authors_page"]
+            task = self.site.generic_post_list_renderer(
+                lang,
+                [],
+                output_name,
+                template_name,
+                kw['filters'],
+                context,
+            )
+            task['uptodate'] = task['uptodate'] + [utils.config_changed(kw, 'nikola.plugins.task.authors:page')]
+            task['basename'] = str(self.name)
+            yield task
+
+    def list_authors_page(self, kw):
+        """Create a global "all authors" page for each language."""
+        yield self._create_authors_page(kw)
+
+    def _get_title(self, author):
+        return author
+
+    def _get_description(self, author, lang):
+        descriptions = self.site.config['AUTHOR_PAGES_DESCRIPTIONS']
+        return descriptions[lang][author] if lang in descriptions and author in descriptions[lang] else None
+
+    def author_page_as_index(self, author, lang, post_list, kw):
+        """Render a sort of index page collection using only this author's posts."""
+        kind = "author"
+
+        def page_link(i, displayed_i, num_pages, force_addition, extension=None):
+            feed = "_atom" if extension == ".atom" else ""
+            return utils.adjust_name_for_index_link(self.site.link(kind + feed, author, lang), i, displayed_i, lang, self.site, force_addition, extension)
+
+        def page_path(i, displayed_i, num_pages, force_addition, extension=None):
+            feed = "_atom" if extension == ".atom" else ""
+            return utils.adjust_name_for_index_path(self.site.path(kind + feed, author, lang), i, displayed_i, lang, self.site, force_addition, extension)
+
+        context_source = {}
+        title = self._get_title(author)
+        if kw["generate_rss"]:
+            # On a author page, the feeds include the author's feeds
+            rss_link = ("""<link rel="alternate" type="application/rss+xml" """
+                        """type="application/rss+xml" title="RSS for author """
+                        """{0} ({1})" href="{2}">""".format(
+                            title, lang, self.site.link(kind + "_rss", author, lang)))
+            context_source['rss_link'] = rss_link
+        context_source["author"] = title
+        indexes_title = kw["messages"][lang]["Posts by %s"] % title
+        context_source["description"] = self._get_description(author, lang)
+        context_source["pagekind"] = ["index", "author_page"]
```

Comment:
I added documentation for this new pagekind as well as all the other existing ones to `docs/theming.txt`.